x-pack/plugins/cloud/server/config.ts

```diff
@@ -33,9 +33,14 @@ const configSchema = schema.object({
   trial_end_date: schema.maybe(schema.string()),
   is_elastic_staff_owned: schema.maybe(schema.boolean()),
   serverless: schema.maybe(
-    schema.object({
-      project_id: schema.string(),
-    })
+    schema.object(
+      {
+        project_id: schema.string(),
+        project_name: schema.maybe(schema.string()),
+      },
+      // avoid future chicken-and-egg situation with the component populating the config
+      { unknowns: 'ignore' }
+    )
```

Comment:
For future additions, this will allow to inject the data from the controller before having the version of Kibana knowing of the new setting is promoted to production